test_util/cluster.py

```diff
@@ -91,10 +89,7 @@ def __init__(
             'All cluster hosts require a private IP. hosts: {}'.format(repr(self.hosts))
         )
 
-        if self.provider == 'onprem':
-            self.check_ssh(self.hosts)
```

Comment:
All hosts are checked because this class is made specifically for facilitating upgrades in onprem deploys, which requires lots of SSH tidbits. In theory this could be done with a daisy chain through a master, but at the moment that would require a bit too much effort or would increase flakiness. Anyway, this check was preserved below in the from_vpc method. The reason it was added there is it was really just required because AWS will say a VPC CF is ready to go before SSH will accept connections, so you need to make sure all hosts are ready. This could also be done in from_hosts, but I would rather we just make this so that its a check we only do for upgrades or for specific provisioning where we know its required